README.md

```diff
@@ -378,10 +378,127 @@ export class VehicleProductComponent extends NgxSubFormRemapComponent<OneVehicle
 
 Our "incoming" object is of type `OneVehicle` but into that component we treat it as a `OneVehicleForm` to split the vehicle (either a `speeder` or `spaceship`) in 2 **separated** properties.
 
+### Dealing with arrays
+
+When your data structure contains array(s), you may want to simply display the values in the view but chances are: You want to bind them to the form.
+
+In that case, working with a `FormArray` is the right way to go and for that, we will take advantage of the remap principles explained in the previous section.
+
+Example:
+
+```ts
+// src/app/main/listing/listing-form/vehicle-listing/crew-members/crew-members.component.ts#L5-L54
+
+interface CrewMembersForm {
+  crewMembers: CrewMember[];
+}
+
+@Component({
+  selector: 'app-crew-members',
+  templateUrl: './crew-members.component.html',
+  styleUrls: ['./crew-members.component.scss'],
+  providers: subformComponentProviders(CrewMembersComponent),
+})
+export class CrewMembersComponent extends NgxSubFormRemapComponent<CrewMember[], CrewMembersForm> {
+  protected getFormControls(): Controls<CrewMembersForm> {
+    return {
+      crewMembers: new FormArray([]),
+    };
+  }
+
+  protected transformToFormGroup(obj: CrewMember[] | null): CrewMembersForm {
+    return {
+      crewMembers: obj ? obj : [],
+    };
+  }
+
+  protected transformFromFormGroup(formValue: CrewMembersForm): CrewMember[] | null {
+    return formValue.crewMembers;
+  }
+
+  public removeCrewMember(index: number): void {
+    this.formGroupControls.crewMembers.removeAt(index);
+  }
+
+  public addCrewMember(): void {
+    this.formGroupControls.crewMembers.push(new FormControl());
```

Comment:
This will result in inconsistent behavior with this form, as this form control has no validators but the hook makes it required. It should really be 

```ts
public addCrewMember(): void {
    this.formGroupControls.crewMembers.push(this.createFormArrayControl((this.formControlNames.crewMembers));
}
```

However, that is going to be a really common use case, so I think we should abstract it into a 

```ts
protected addFormArrayElement(formControlName, initialValue = null): void;
```
in the parent, which means this example would simply be 

```
public addCrewMember(): void {
   this.addFormArrayElement(this.formControlNames.crewMembers);
}
```

or even simpler in the html
```html
  <button
    mat-raised-button
    data-btn-add-crew-member
    color="primary"
    class="add-crew-member"
    (click)="addFormArrayElement('crewMember')"
    [disabled]="formGroup.disabled"
  >
    Add a crew member
  </button>

```

Reply:
I believe this is now fixed @zakhenry, Isn't it?

Reply:
Correct, I had a go at writing a nice utility for creating/deleting array entities but when I switched to interface syntax I realised it was better to just have the developer do it themselves as their needs may wildly differ.